data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java

```diff
@@ -346,6 +385,7 @@ public static ConnectionStringBuilder createWithAadApplicationCredentials(String
         }
 
         ConnectionStringBuilder csb = new ConnectionStringBuilder();
+        csb.aadFederatedSecurity = true;
```

Comment:
Is there a no auth option in Java?

Reply:
yes

Reply:
Recently added, but haven't been released outside of alpha.

I think I'll align it be the same logic we have in c# and python:

- If Fed=True and no auth specified - default to prompt
- If Fed is not given, but we have some other auth - assume Fed=true and use the auth
- If Fed=False and no auth given, assume no auth mode.

Sounds ok?